src/odemis/driver/orsay.py

```diff
@@ -1389,3 +1440,1097 @@ def terminate(self):
+            target = self._VA_to_parameter_value(goal)
+            self._parameters[0].Target = target
+            logging.debug("Changing %s to %s." % (self._parameters[0].Name, str(target)))
+
+        return goal
+
+    def _VA_to_parameter_value(self, va_value):
+        """
+        Converts a value of the VA to its corresponding value for the parameter
+        va_value is the value of the VA
+        returns the corresponding value of the parameter
+        """
+        if self._conversion is not None:  # if a conversion dict is supplied
+            try:
+                return self._conversion[va_value]
+            except KeyError:
+                logging.warning("Conversion dictionary does not contain key %s. Sticking to value %s" %
+                                (str(va_value), str(va_value)))
+        elif self._factor is not None and self._va_value_type == float:
+            return va_value / self._factor
+        return va_value
+
+    def _parameter_to_VA_value(self, par_value):
+        """
+        Converts a value of the parameter to its corresponding value for the VA
+        par_value is the value of the parameter
+        returns the corresponding value of the VA
+        """
+        if self._conversion is not None:  # if a conversion dict is supplied
+            for key, value in self._conversion.items():
+                if value == type(value)(par_value):
+                    return key
+
+        if self._va_value_type == float:
+            new_value = float(par_value)
+            if self._factor is not None:
+                new_value *= self._factor
+            return new_value
+        elif self._va_value_type == int:
+            return int(par_value)
+        elif self._va_value_type == bool:
+            return par_value in {True, "True", "true", 1, "1", "ON"}
+        else:
+            raise NotImplementedError("Handeling of VA's of type %s is not implemented for OrsayParameterConnector."
+                                      % self._va_type_name)
+
+
+class FIBDevice(model.HwComponent):
+    """
+    Represents the Focused Ion Beam (FIB) device from Orsay Physics. Contains generic device properties and settings
+    """
+
+    def __init__(self, name, role, parent, **kwargs):
+        """
+        Defines the following VA's and links them to the callbacks from the Orsay server:
+        • interlockTriggered: BooleanVA
+        • valveOpen: BooleanVA
+        • gunPumpOn: BooleanVA
+        • columnPumpOn: BooleanVA
+        • gunPressure: FloatContinuous, readonly, unit="Pa", range=(0, 11e4)
+        • columnPressure: FloatContinuous, readonly, unit="Pa", range=(0, 11e4)
+        • compressedAirPressure: FloatContinuous, readonly, unit="Pa", range=(0, 5e6)
+        """
+
+        super().__init__(name, role, parent=parent, **kwargs)
+
+        self._gunPump = None
+        self._columnPump = None
+        self._interlockHVPS = None
+        self._interlockChamber = None
+        self._valve = None
```

Comment:
I'll add a comment. I define them here, because otherwise Pycharm complains that I am defining attributes in a place other than init and that's not nice. Related to what you said: If someone looks for the source, they'll start searching in init.